### PR TITLE
Added simple example of using TFeat for patch description

### PR DESCRIPTION
**Pull request by ducha-aiki**
Also added stats file (mean, var) from pnnet repository and graffity refpng from hpatches dataset as example image

### CODE REVIEW

File: extract_desciptors_from_hpatch_file.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python2 -utt
+# -*- coding: utf-8 -*-
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py]: are this two lines really needed?

File: extract_desciptors_from_hpatch_file.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import sys
+import numpy as np
+import lutorpy as lua
+import torchfile
+import os
+import sys
+import time
+#sys.path.insert(0, '/home/ubuntu/dev/opencv-3.1/build/lib')
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py]: remove relative paths

File: extract_desciptors_from_hpatch_file.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import cv2
+TFEAT_PATCH_SIZE = 32
+TFEAT_DESC_SIZE = 128
+TFEAT_BATCH_SIZE = 1000
+STATS_FNAME = 'nets/stats.liberty.t7'
+MODEL_FNAME = 'nets/tfeat_liberty_margin_star.t7'
+
+try:
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py:20]: Could you please add something like that:
```python
try:
    import torchfile
except ImportError:
    torchfile = None
```
and then check with:
```python
if torchfile is None:
    raise ImportError("Please, do pip install torchfile.")
```

File: extract_desciptors_from_hpatch_file.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    STD = stats['sigma']
+except:
+    print "Please, do pip install torchfile, and ensure that there is nets/stats.liberty.t7 file"
+    sys.exit(1)
+
+def preprocess_patch(patch):
+    out = cv2.resize(patch, (TFEAT_PATCH_SIZE, TFEAT_PATCH_SIZE)).astype(np.float32) / 255;
+    out -= MEAN
+    out /= STD
+    return out.reshape(1, TFEAT_PATCH_SIZE, TFEAT_PATCH_SIZE)
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py]: add blankline space

File: extract_desciptors_from_hpatch_file.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    MEAN = stats['mi']
+    STD = stats['sigma']
+except:
+    print "Please, do pip install torchfile, and ensure that there is nets/stats.liberty.t7 file"
+    sys.exit(1)
+
+def preprocess_patch(patch):
+    out = cv2.resize(patch, (TFEAT_PATCH_SIZE, TFEAT_PATCH_SIZE)).astype(np.float32) / 255;
+    out -= MEAN
+    out /= STD
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py]:
```python 
out = (out - MEAN) / STD
```

File: extract_desciptors_from_hpatch_file.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    # split patches into batches
+    patches_t = patches_t._split(TFEAT_BATCH_SIZE)
+    descriptors = []
+    for i in range(int(np.ceil(float(num) / TFEAT_BATCH_SIZE))):
+        # infere Torch network
+        prediction_t = net._forward(patches_t[i]._cuda())
+        # Cast TorchTensor to NumpyArray and append to results
+        prediction = prediction_t.asNumpyArray()
+        # add the current prediction to the buffer
+        #print prediction.shape
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py]: remove comments

File: extract_desciptors_from_hpatch_file.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        descriptors.append(prediction)
+    out =  np.concatenate(descriptors)
+    return out.reshape(num, TFEAT_DESC_SIZE)
+
+if __name__ == '__main__':
+    try:
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py]:
```python
if len(sys.argv) < 2:
    print('Wrong input format. Try ./extract_desciptors_from_hpatch_file.py imgs/ref.png ref.TFEAT')
    sys.exit(1)

input_img_fname = sys.argv[1]
output_fname = sys.argv[2]

File: extract_desciptors_from_hpatch_file.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        input_img_fname = sys.argv[1]
+        output_fname = sys.argv[2]
+    except:
+        print "Wrong input format. Try ./extract_desciptors_from_hpatch_file.py imgs/ref.png ref.TFEAT"
+        sys.exit(1)
+    t = time.time()
+    image = cv2.imread(input_img_fname,0) #hpatch image is patch column 65*n x 65
+    h,w = image.shape
+    n_patches = h/w
+    print n_patches ," patches to describe in",input_img_fname
```
Comment (edgarriba) [inline extract_desciptors_from_hpatch_file.py]:
```python
print('{0} patches to describe in {1}'.format(n_patches, input_img_fname))
```

**Review by edgarriba — CHANGES_REQUESTED** [review]

**Comment by edgarriba** [comment]
@ducha-aiki why do you include ref.png?

File: extract_desciptors_from_hpatch_file.py (hunk 6 above)
Comment (ducha-aiki) [reply extract_desciptors_from_hpatch_file.py]: I will remove them, but they are taken from https://github.com/vbalnt/tfeat/blob/master/tfeat_demo.py#L97
If they are not needed, why do you included them?

**Comment by ducha-aiki** [comment]
@edgarriba because I strongly believe that any example "how to use something" should run "as is".

**Comment by ducha-aiki** [comment]
Changes applied

**Comment by vbalnt** [comment]
Thanks, this is also related to issue #1. 

The code merged above, shows how to do the proper normalisation, to mimic the one that is done in the torch code (and more importantly, the one that the weights were trained with)
